### PR TITLE
Add a framework to save/load operators.

### PR DESCRIPTION
**Pull request by Spaceenter**

### CODE REVIEW

**Comment by Spaceenter** [comment]
@babbush

File: src/fermilib/utils/__init__.py [hunk 1]
```diff
@@ -30,7 +30,8 @@
 from ._molecular_data import MolecularData, periodic_table
 
 from ._operator_utils import (eigenspectrum, commutator,
-                              count_qubits, is_identity)
+                              count_qubits, is_identity,
+                              save_operator, load_operator, get_file_path)
```
Comment (babbush) [inline src/fermilib/utils/__init__.py]: can you please make the imports in alphabetical order (i.e. get_file_path, load_operator, save_operator)?

File: src/fermilib/utils/_operator_utils.py [hunk 2]
```diff
@@ -102,3 +111,102 @@ def commutator(operator_a, operator_b):
...
+    Raises:
+        OperatorUtilsError: Not saved, file already exists.
+        TypeError: Operator of invalid type.
+    """
+    file_path = get_file_path(file_name, data_directory)
+
+    if os.path.isfile(file_path):
+        raise OperatorUtilsError("Not saved, file already exists.")
+
+    if isinstance(operator, FermionOperator):
```
Comment (babbush) [inline src/fermilib/utils/_operator_utils.py]: Would you mind adding a "not implemented error" specifically when somebody tries to save an InteractionOperator or InteractionRDM? Eventually (but not now) we may want to include that.

File: src/fermilib/utils/_operator_utils.py [hunk 3]
```diff
@@ -102,3 +111,102 @@ def commutator(operator_a, operator_b):
...
+            [{'k': k, 'r': v.real, 'i': v.imag}
+                for k, v in operator.terms.iteritems()],
+            ensure_ascii=False).encode('utf8'))
+
+
+def load_operator(file_name=None, data_directory=None):
+    """Load FermionOperator or QubitOperator from hdf5 file.
+
+    Args:
+        file_name: The name of the saved hdf5 file. If not provided, the unix
```
Comment (babbush) [inline src/fermilib/utils/_operator_utils.py]: I'm a little confused about what the default file name is when loading. I feel like the filename should not be optional for the load function. Do you agree?

File: src/fermilib/utils/_operator_utils_test.py [hunk 4]
```diff
@@ -14,12 +14,15 @@
...
+import os
 import unittest
 
+from fermilib.config import *
 from fermilib.ops import *
 from fermilib.transforms import jordan_wigner, get_interaction_operator
 from fermilib.utils import (eigenspectrum, commutator,
-                            count_qubits, is_identity)
+                            count_qubits, is_identity,
+                            save_operator, load_operator, get_file_path)
```
Comment (babbush) [inline src/fermilib/utils/_operator_utils_test.py]: alphabetical imports

File: src/fermilib/utils/_operator_utils_test.py [hunk 5]
```diff
@@ -14,12 +14,15 @@
...
 
 from projectq.ops import QubitOperator
```
Comment (babbush) [inline src/fermilib/utils/_operator_utils_test.py]: Can you possibly add another test? Maybe one that tests the save when file name is "None". We are currently trying to only accept PRs that raise the test coverage or keep it the same and this one lowers test coverage by 0.06%.

**Comment by Spaceenter** [comment]
@babbush 
Cleaned up according to our comments.

**Comment by babbush** [comment]
👍

**Comment by babbush** [comment]
I will merge as soon as tests complete. Travis seems to not be running though.
